docs/source/developers/guide/step_by_step/building.rst

```diff
@@ -155,6 +155,22 @@ Building other Arrow libraries
      `R developer docs <https://arrow.apache.org/docs/r/articles/developing.html>`_.
 
 
+     **Reinstalling R package and running 'make clean'**
+
+     When you will make changes to the Arrow C++ part of the code, also
+     called libarrow, you will need to:
+
+     #. reinstall libarrow
+     #. run ``make clean`` or ``git clean -fdx .``
```

Comment:
``make clean`` and ``git clean -fdx .`` do totally different things to each other so I don't think the phrasing is quite right here. I've never run ``git clean -fdx .`` while doing dev work (it could be useful, but isn't essential given I just had to google what it was)